src/libpmemfile/CMakeLists.txt

```diff
@@ -31,7 +31,9 @@
 
 set(SOURCES path_resolve.c preload.c)
 
-add_c_flag(-Wmissing-field-initializers)
+if("${CMAKE_C_COMPILER_ID}" MATCHES "Clang")
+	add_c_flag(-Wno-missing-field-initializers)
+endif()
```

Comment:
Please move this and the same code (with comment) in libpmemfile-posix/CMakeLists.txt to a new function in main CMakeLists.txt and call it from both places.